baseline/pytorch/classify/model.py

```diff
@@ -127,68 +127,154 @@ def predict(self, batch_dict, raw=False, dense=False, **kwargs):
+    """
 
-    def init_pool(self, dsz, **kwargs):
-        filtsz = kwargs['filtsz']
+    def init_pool(self, input_dim: int, **kwargs) -> BaseLayer:
+        """Do parallel convolutional filtering with varied receptive field widths, followed by max-over-time pooling
+
+        :param input_dim: Embedding output size
+        :param kwargs: See below
+
+        :Keyword Arguments:
+        * *cmotsz* -- (``int``) The number of convolutional feature maps for each filter
+            These are MOT-filtered, leaving this # of units per parallel filter
+        * *filtsz* -- (``list``) This is a list of filter widths to use
+
+        :return: A pooling layer
+        """
         cmotsz = kwargs['cmotsz']
-        return WithoutLength(WithDropout(ParallelConv(dsz, cmotsz, filtsz, "relu", input_fmt="bth"), self.pdrop))
+        filtsz = kwargs['filtsz']
+        return WithoutLength(WithDropout(ParallelConv(input_dim, cmotsz, filtsz, "relu", input_fmt="bth"), self.pdrop))
 
 
 @register_model(task='classify', name='lstm')
-class LSTMModel(ClassifierModelBase):
+class LSTMModel(EmbedPoolStackClassifier):
+    """A simple single-directional single-layer LSTM. No layer-stacking.
+    """
 
-    def init_pool(self, dsz, **kwargs):
+    def init_pool(self, input_dim: int, **kwargs) -> BaseLayer:
+        """LSTM with dropout yielding a final-state as output
+
+        :param input_dim: The input word embedding depth
+        :param kwargs: See below
+
+        :Keyword Arguments:
+        * *rnnsz* -- (``int``) The number of hidden units (defaults to `hsz`)
+        * *rnntype/rnn_type* -- (``str``) The RNN type, defaults to `lstm`, other valid values: `blstm`
+        * *hsz* -- (``int``) backoff for `rnnsz`, typically a result of stacking params.  This keeps things simple so
+          its easy to do things like residual connections between LSTM and post-LSTM stacking layers
+
+        :return: A pooling layer
+        """
         unif = kwargs.get('unif')
         hsz = kwargs.get('rnnsz', kwargs.get('hsz', 100))
         if type(hsz) is list:
             hsz = hsz[0]
         weight_init = kwargs.get('weight_init', 'uniform')
         rnntype = kwargs.get('rnn_type', kwargs.get('rnntype', 'lstm'))
         if rnntype == 'blstm':
-            return BiLSTMEncoderHidden(dsz, hsz, 1, self.pdrop, unif=unif, batch_first=True, initializer=weight_init)
-        return LSTMEncoderHidden(dsz, hsz, 1, self.pdrop, unif=unif, batch_first=True, initializer=weight_init)
+            return BiLSTMEncoderHidden(input_dim, hsz, 1, self.pdrop, unif=unif, batch_first=True, initializer=weight_init)
+        return LSTMEncoderHidden(input_dim, hsz, 1, self.pdrop, unif=unif, batch_first=True, initializer=weight_init)
 
     def make_input(self, batch_dict):
+        """In PyTorch, we typically set the lengths to sort descending and permute the batch accordingly
```

Comment:
This length permutation stuff has since been added to the base `make_input`. I think we can remove this implementation in the subclass